cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/ExternalSparkProgram.java

```diff
@@ -113,4 +119,28 @@ public void destroy() {
       }
     }
   }
+
+  /**
+   * Gets all the plugins used in the program through the {@link PipelinePhase} and calls
+   * {@link #usePluginClass(String, String, String, PluginProperties, PluginSelector)} explicitly to register their
+   * usage so that they are accessible in the current configurer obtained through {@link #getConfigurer()}
```

Comment:
the reason to register them here isn't to make them available to the configurer. It's to allow CDAP to quickly fail if somebody runs this program with plugins that are incompatible with the profile.

Reply:
This got moved. I have added appropriate doc there.